### PR TITLE
fix: remove cdk re-exports from @angular/material

### PR DESCRIPTION
**Pull request by jelbourn**
BREAKING CHANGE: `@angular/material` no longer re-exports symbols from
`@angular/cdk`.

### CODE REVIEW

File: src/lib/core/rxjs/rx-chain.spec.ts [hunk 1]
```diff
@@ -1,41 +0,0 @@
-import {Observable} from 'rxjs/Observable';
```
Comment (crisbeto) [inline src/lib/core/rxjs/rx-chain.spec.ts]: It seems like this removes the `rx-chain` tests completely.

Comment (jelbourn) [reply src/lib/core/rxjs/rx-chain.spec.ts]: They were already copied to `cdk` earlier but not removed from `lib`

**Comment by jelbourn** [comment]
Caretaker note: I have an internal change to go in before this one to prevent it from breaking anyone

**Comment by jelbourn** [comment]
Fixing unit tests, but that shouldn't block presumit

**Comment by jelbourn** [comment]
Unit tests were actually broken in #6822, and I sent a fix in #7117

**Comment by mmalerba** [comment]
needs rebase

**Comment by jelbourn** [comment]
Rebased

**Pull request merged**